src/aead/aead.rs

```diff
@@ -25,6 +25,82 @@
 //! [AEAD]: http://www-cse.ucsd.edu/~mihir/papers/oem.html
 //! [`crypto.cipher.AEAD`]: https://golang.org/pkg/crypto/cipher/#AEAD
 
+//!
+//! **Example (if you don't know what you are doing, read this first)**
+//!
+//!```rust
+//!
+//!extern crate ring;
+//!
+//!use ring::aead::*;
+//!use ring::pbkdf2::*;
+//!use ring::rand::SystemRandom;
+//!
+//!fn main() {
+//!    // The password will be used to generate a key and it is normally used when
+//!    // it is generated by the user. If you need a key which will be stored in a file you
+//!    // generate it directly.
+//!    let password = b"nice password";
+//!
+//!    // Usually the salt has some random data and something that relates to the user
+//!    // like an username and it must be 8 bytes (64 bits)
+//!    let salt = [0, 1, 2, 3, 4, 5, 6, 7];
+//!
+//!    // Keys must have 32 bytes and if you want to use an user password for it you have
+//!    // to use an algorithm to derive it (in this case pbkdf2).
+//!    let mut key = [0; 32];
+//!    derive(&HMAC_SHA256, 100, &salt, &password[..], &mut key);
+//!
```

Comment:
I'd cut off the example here (move the part above elsewhere) and show how to generate a random key here, instead. Many users of other libraries (not ring) fail to generate random keys and hard-code something instead.

Next to the line where the generation happens, you can then explain that the key can also be derived from the user's password or sth. like that and then point to the part of the code above (moved to a second code-block here or to the `pbkdf2` module).

Added benefit: this code sample becomes shorter (it's a bit too long and users might skip it for that reason).